pkg/graph/resolvers/system_intake.go

```diff
@@ -74,6 +77,24 @@ func UpdateSystemIntakeContact(
 	}, nil
 }
 
+// UpdateSystemIntakeRequestType updates a system intake's request type and returns the updated intake.
+// It will return an error if the intake is not found by the ID, or the update fails for any reason.
+func UpdateSystemIntakeRequestType(ctx context.Context, store *storage.Store, systemIntakeID uuid.UUID, newType models.SystemIntakeRequestType) (*models.SystemIntake, error) {
+	// Fetch intake by ID
+	intake, err := store.FetchSystemIntakeByID(ctx, systemIntakeID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Update request type and set UpdatedAt
+	intake.RequestType = newType
+	intake.UpdatedAt = helpers.PointerTo(time.Now())
+
+	// Save intake to DB
+	savedIntake, err := store.UpdateSystemIntake(ctx, intake)
```

Comment:
by no means for this PR: just thinking out loud - is there a world where we do not need to fetch the intake and can instead pass arbitrary fields and the given ID to the updater?

Reply:
Definitely - I took an (admittedly) quick approach for this to unblock Adam's work on the frontend (since this mutation doesn't exist, and it kinda needed to be made to allow this flow to function as expected!)

We could replace this with a single `store` method that just takes an ID and a Type!

Reply:
Definitely! It would be a large refactor and certainly its own ticket/PR/discussion